pkg/pods/manifest_parser.go

```diff
@@ -74,6 +97,14 @@ func (manifest *PodManifest) Write(out io.Writer) error {
 }
 
 func (manifest *PodManifest) Bytes() ([]byte, error) {
+	if manifest.signature != nil {
+		// if it's signed, we must recycle the original content to preserve the
+		// signature's validity. remarshaling it might change the exact text of
```

Comment:
No, hash ordering is random. The reason that we can't remarshal is because the originally signed YAML might be different from the marshalled YAML.

For example I compose a manifest with this exact text:
```yaml
---
# author: sjung
id: web
launchables:
    web:
        launchable_type: hoist
        launchable_id: web
        location: /home/sjung/web.tar.gz
```

Note that the indentation is 4 spaces (YAML standard indentation is 2) and there's also a comment. These are superficial features. If I remarshal the YAML all of them will disappear. The indentation will go back to 2 spaces and there will be no comments, because they aren't preserved when you parse the YAML.

Now I sign the manifest above. My signature is tied to the exact text that I had when I made the signature (which is the whole point). If I remarshal the YAML, then the text will be slightly different, and the signature will no longer match against the resulting manifest. So the original manifest has to be preserved for later signature validation.